doc/Data Plane Code Generation/Add_ProtocolMethods_In_HLCGenerated_SDK.md

```diff
@@ -0,0 +1,310 @@
+# Add Protocol Methods In HLC Generated SDK
+
+In this tutorial, we will step through the process of generating protocol methods in HLC SDK. Protocol methods will be generated in internal RestClient based on the `protocol-method-list` config.
+
+## Prerequisites
+
+You should have HLC SDK generated in `azure-sdk-for-net/sdk/<service name>/<package name>/src/Generated` folder.
+
+## Add protocol methods in generated RestClient
+
+In order to generate protocol methods in HLC generated RestClient, you'd have to define a `protocol-method-list` config in your `autorest.md` file. You can find `autorest.md` file path in `azure-sdk-for-net/sdk/<service name>/<package name>/src/autorest.md`. This config takes a list of full `operationId` defined in it's swagger definition. 
+
+For this tutorial we are going to use `Azure.Data.Tables` SDK. You can find swagger definition for Table service [here](https://github.com/Azure/azure-rest-api-specs/blob/2df8b07bf9af7c96066ca4dda21b79297307d108/specification/cosmos-db/data-plane/Microsoft.Tables/preview/2019-02-02/table.json), `autorest.md` file [here](https://github.com/azure-sdk/azure-sdk-for-net/blob/17debdffe16df01ae196579c91ea22e77eddc96a/sdk/tables/Azure.Data.Tables/src/autorest.md) and `Generated` folder [here](https://github.com/azure-sdk/azure-sdk-for-net/tree/17debdffe16df01ae196579c91ea22e77eddc96a/sdk/tables/Azure.Data.Tables/src/Generated).
+
+* ### Generated code before:
+
+In Generated folder, you will find `TableRestClient.Delete` and `ServiceRestClient.SetProperties` methods are already generated as per default HLC configuration.
+
+**(Generated/TableRestClient.cs)**:
```

Comment:
Would it be possible to show the API listing before and after, rather than the full generated code?  I think it is more concise and illustrates what's happening at a conceptual level a little better.  We can keep the full generated code hidden in the details blocks -- I really like the diff below where you see the protocol methods added.

You can use the API format we use in the library API definitions (e.g. here: https://github.com/azure-sdk/azure-sdk-for-net/blob/17debdffe16df01ae196579c91ea22e77eddc96a/sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs#L140)

I think it would be nice to see something like

Before:
```csharp
    public partial class TableClient
    {
        public TableClient(string connectionString, string tableName) { }
        // ...
        public virtual Response SetProperties(TableServiceProperties properties, CancellationToken cancellationToken = default(CancellationToken)) { throw null; }
        public virtual Task<Response> SetPropertiesAsync(TableServiceProperties properties, CancellationToken cancellationToken = default(CancellationToken)) { throw null; }
        // ...
```

After

```csharp
    public partial class TableClient
    {
        public TableClient(string connectionString, string tableName) { }
        // ...
        public virtual Response SetProperties(TableServiceProperties properties, CancellationToken cancellationToken = default(CancellationToken)) { throw null; }
        public virtual Task<Response> SetPropertiesAsync(RequestContent properties, RequestContext context = null) { throw null; }
        public virtual Response SetProperties(TableServiceProperties properties, CancellationToken cancellationToken = default(CancellationToken)) { throw null; }
        public virtual Task<Response> SetPropertiesAsync(TableServiceProperties properties, CancellationToken cancellationToken = default(CancellationToken)) { throw null; }
        // ...
```

With the nice diff highlighting showing what's added.

Reply:
Updated!